### ISSUE TITLE
Nuitka - issue with cvxpy

### ISSUE THREAD

**Issue by dmitrii-erkin** [Nuitka/Nuitka#1308]
Hello,

I am facing an issue with cvxpy:

```

import cvxpy as cp

y = cp.Variable((1, 106), boolean=False, nonneg=True)
print(y[:1, 0])
```

gives me

```
~/sandbox/fleet-scheduling/src$ ./cvxpy_test.dist/cvxpy_test
Traceback (most recent call last):
  File "~/sandbox/fleet-scheduling/src/cvxpy_test.dist/cvxpy_test.py", line 4, in <module>
  File "~/sandbox/fleet-scheduling/src/cvxpy_test.dist/cvxpy/expressions/expression.py", line 424, in __getitem__
  File "~/sandbox/fleet-scheduling/src/cvxpy_test.dist/cvxpy/atoms/affine/index.py", line 54, in __init__
  File "~/sandbox/fleet-scheduling/src/cvxpy_test.dist/cvxpy/atoms/atom.py", line 51, in __init__
  File "~/sandbox/fleet-scheduling/src/cvxpy_test.dist/cvxpy/atoms/affine/index.py", line 80, in shape_from_args
  File "~/sandbox/fleet-scheduling/src/cvxpy_test.dist/cvxpy/utilities/key_utils.py", line 188, in shape
TypeError: unsupported operand type(s) for /: 'int' and 'NoneType'
```

* Please check out if the develop version of Nuitka works better for you.

I am using 0.6.17, because nuitka 0.6.18 is incompatible with hinted compilation utils.

* Nuitka version, full Python version and Platform (Windows, OSX, Linux ...)

  > python -m nuitka --version
```

0.6.17
Commercial: None
Python: 3.9.9 (main, Nov 16 2021, 03:08:02) 
Executable: /usr/bin/python
OS: Linux
Arch: x86_64
```

* How did you install Nuitka and Python
`python -m pip install nuitka`

* The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g.
  `numpy`, you are expected to shorten this to the relevant ones.

  > python -m pip freeze

```
cvxopt==1.2.7
cvxpy==1.1.17
cylp==0.91.4
Cython==0.29.25
```

Any help is greatly appreciated. Thank you.

**Comment by kayhayen** [comment]
Note: Not using 0.6.18 is not acceptable for non-commercial users to get support. Any fix will be in there only.

**Comment by kayhayen** [comment]
That said, I managed to reproduce the issue on 0.6.18 indeed. I didn't find what might be going on there, I suspect there might be some data files missing. With anti-bloat plugin its relatively manageable, and it seems to not pull in too much modules, however, that package seems crapy quality:

```
>>> import cvxpy.utilities.key_utils
>>> cvxpy.utilities.key_utils
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'cvxpy.reductions.utilities' has no attribute 'key_utils'
```

Which seems to indicate very bad code. I will defer this one until I manage to manage failed file access detections for standalone. I think by default, standalone should detect failed open, empty listdir, etc. operations and therefore make it easy to debug these, that code is a bit too hard for me to want to debug right now without that tool.

**Comment by dmitriierkineroad** [comment]
> 

Hi @kayhayen thank you very much for your swift response. Regarding using 0.6.18, are you aware of any plans to make [utils](https://github.com/Nuitka/NUITKA-Utilities/tree/master/hinted-compilation) compatible with this latest version, Thank you,

**Comment by kayhayen** [comment]
There is no such plan, but Nuitka will gain superior functionality called `--pgo-python`. Due to `anti-bloat` plugin I do consider it obsolete anyway. Nobody is working on it anymore anyway.

**Comment by kayhayen** [comment]
So, I am compiling this without issues, but the runtime problem remains the originally reported one. I am checking for hidden dependencies, some modules fail to import. We got the import tracing from our "lib", that I haven't had to use in a while, but this seems pretty complex. The crash is most probably some type not registered with something numpy.

**Comment by kayhayen** [comment]
I repaired the import tracing, such that it actually allows me to compare things between the standalone build and the Python build, but I only found minor line number differences in the import chain, but modules imported and the success of it were the same, so it doesn't appear to be that. I guess, I will have to go back to the data files theory.

**Comment by kayhayen** [comment]
It's also not data files or DLLs. I started to trace values, and it's peculiar, but `slice` object value with a step `None` is causing the error, and `None` is the default for a slice step.

So very likely this is caused by this optimization:

```
        if (
            self.subnode_step.isExpressionConstantNoneRef()
            or self.subnode_step.getIndexValue() == 1
        ):
            return trace_collection.computedExpressionResult(
                wrapExpressionWithSideEffects(
                    old_node=self,
                    new_node=ExpressionBuiltinSlice2(
                        start=self.subnode_start,
                        stop=self.subnode_stop,
                        source_ref=self.source_ref,
                    ),
                    side_effects=self.subnode_step.extractSideEffects(),
                ),
                "new_expression",
                "Reduce 3 argument slice object creation to two argument form.",
            )
```

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by kayhayen** [comment]
This is part of the hotfix release 1.7.4 that I just made.
